pkg/rtp/packet.go

```diff
@@ -169,7 +170,7 @@ func (p *Packet) Marshal() ([]byte, error) {
 	rawPacket[1] |= p.PayloadType
 
 	binary.BigEndian.PutUint16(rawPacket[seqNumOffset:], p.SequenceNumber)
-	binary.BigEndian.PutUint32(rawPacket[timestampOffset:], p.Timestamp)
+	binary.BigEndian.PutUint32(rawPacket[timestampOffset:], uint32(p.Timestamp))
```

Comment:
This conversion (and the marshall version, above) may not be correct by virtue of the fact that the RTP timestamp may not be in the same units for every packet type. for instance, H264 requires a timestamp clocked at 90 Khz, while many audio payload types (at least μlaw and G.722) have a clock rate of 8 Khz. maybe you can add the clock rate as a property of the payloader, and pass the payloader to the marshal and unmarshal members? At any rate, this needs to be more complicated then it is.

Reply:
another idea is to have time.Duration<->uint32 conversion functions themselves as members of the payloader interface.